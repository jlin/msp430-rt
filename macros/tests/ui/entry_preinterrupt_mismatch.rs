#![no_main]

use msp430::interrupt::CriticalSection;
use msp430_rt_macros::entry;

fn init(cs: CriticalSection) -> u32 {
    32
}

#[entry(interrupt_enable(pre_interrupt = init))]
fn main(_i: bool) -> ! {
    loop {}
}
