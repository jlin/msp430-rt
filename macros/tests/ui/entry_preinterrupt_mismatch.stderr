error[E0308]: mismatched types
  --> tests/ui/entry_preinterrupt_mismatch.rs:10:1
   |
10 | #[entry(interrupt_enable(pre_interrupt = init))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `u32`
   |
   = note: this error originates in the attribute macro `entry` (in Nightly builds, run with -Z macro-backtrace for more info)
